packages/api/internal/cache/instance/instance.go

```diff
@@ -39,13 +39,13 @@ type InstanceCache struct {
 
 	logger *zap.SugaredLogger
 
-	counter   metric.Int64UpDownCounter
-	analytics analyticscollector.AnalyticsCollectorClient
+	counter   *metric.Int64UpDownCounter
```

Comment:
I think I understand the dereferences now, but it gets more complicated. These fields are interfaces and interfaces in Go are pointers (to the underlying implementation, struct), so there is rarely a point of using pointer to an interface.
The default Go syntax of dereferencing pointers when accessing fields plays directly into this.